### ISSUE TITLE
I cannot understand the loss function

### ISSUE THREAD

**Issue by mks0601** [weiliu89/caffe#299]
In the paper, p seems category of object(from 1 to 20 in pascal voc, except background class)
So, for positive default boxes, obtain prob of each class by softmax and class negative log likelihood is loss function.
However, for negative default boxes, the prob of neg is annotated as c^hat_0. Is this prob comes from above softmax layer?

I implemented SSD300 and the confusion matrix says most of the matched default box and negative samples are classified to background class. Any help?

**Comment by weiliu89** [comment]
@mks0601 Take VOC for example, p is in [0, 20] where 0 is for background class and [1, 20] is for 20 VOC classes. Confidence loss is a combination of all matched positive samples and all mined hard negatives. Localization loss is only from matched positive samples.
